### DISCUSSION TITLE
What is the correct way to do cross references within a notebook and/or across notebooks in a book?

### DISCUSSION THREAD

**Discussion by mwcraig** [executablebooks/meta#1036] [Q&A] [answered]
After about a year away from updating a jupyter book I maintain I'm trying to update to jupyter-book 0.15. At first I was getting lots of errors along the lines of `WARNING: 'myst' reference target not found` for cross-references between notebooks or within a single notebook.

All of my content is in `.ipynb` files.

After reading https://github.com/executablebooks/jupyter-book/issues/1657 I thought the issue was with the version of `myst` I had, but that wasn't it.

Eventually I found [this in the myst docs](https://myst-parser.readthedocs.io/en/latest/syntax/cross-referencing.html#implicit-targets) which seemed promising, but it took me a long while to turn on `myst_heading_anchors`, which I did by modifying the source code. 

Now my cross-references work in the generated book, but the links don't work if I run the notebooks in JupyterLab. 

That was not surprising, since myst wants github-style slugs.

**I'd really like some way of doing links that works in the book HTML and when I run the notebooks in JupyterLab. Is that possible? How?**

**Comment by rowanc1** [comment, marked as answer]
It is possible that the `jupyterlab-myst` extension might work for your use case. The links and cross-references in the notebooks do open natively as links to notebooks. See the repo here:
https://github.com/executablebooks/jupyterlab-myst

**Reply by mwcraig** [reply]
Thanks -- this mostly does the trick. It opens the correct notebook though it doesn't jump to the right heading in the notebook that is the target of the link.
